multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/heancms/HeanCms.kt

```diff
@@ -230,30 +260,72 @@ abstract class HeanCms(
     }
 
     override fun mangaDetailsParse(response: Response): SManga {
-        val result = runCatching { response.parseAs<HeanCmsSeriesDto>() }
-        val seriesDetails = result.getOrNull()?.toSManga(apiUrl, coverPath, fetchAllTitles)
+        val mangaStatus = response.request.url.fragment?.toIntOrNull() ?: SManga.UNKNOWN
+
+        val result = runCatching {
+            if (fetchAllTitlesStrategy == FetchAllStrategy.SEARCH_EACH) {
+                val originalSlug = response.request.url.fragment!!
+                val searchResult = response.parseAs<List<HeanCmsSearchDto>>()
+                searchResultToSeries(originalSlug, searchResult)
+            } else {
+                response.parseAs<HeanCmsSeriesDto>()
+            }
+        }
+
+        val seriesDetails = result.getOrNull()?.toSManga(apiUrl, coverPath, fetchAllTitlesStrategy)
             ?: throw Exception(intl.urlChangedError(name))
 
         return seriesDetails.apply {
             status = status.takeUnless { it == SManga.UNKNOWN }
-                ?: response.request.url.fragment?.toIntOrNull() ?: SManga.UNKNOWN
+                ?: mangaStatus
         }
     }
 
     override fun chapterListRequest(manga: SManga): Request = mangaDetailsRequest(manga)
 
     override fun chapterListParse(response: Response): List<SChapter> {
-        val result = response.parseAs<HeanCmsSeriesDto>()
-        val seriesSlug = response.request.url.pathSegments.last()
+        val result = if (fetchAllTitlesStrategy == FetchAllStrategy.SEARCH_EACH) {
+            val originalSlug = response.request.url.fragment!!
+            val searchResult = response.parseAs<List<HeanCmsSearchDto>>()
+            searchResultToSeries(originalSlug, searchResult)
+        } else {
+            response.parseAs<HeanCmsSeriesDto>()
+        }
+
         val currentTimestamp = System.currentTimeMillis()
 
         return result.chapters.orEmpty()
             .filterNot { it.price == 1 }
-            .map { it.toSChapter(seriesSlug, dateFormat) }
+            .map { it.toSChapter(result.slug, dateFormat) }
             .filter { it.date_upload <= currentTimestamp }
             .reversed()
     }
 
+    private fun searchResultToSeries(originalSlug: String, searchResult: List<HeanCmsSearchDto>): HeanCmsSeriesDto {
+        val mangaSlug = searchResult
+            .filter { it.type == "Comic" }
+            .map { it.slug }
+            .find { it.replace(TIMESTAMP_REGEX, "") == originalSlug }
+            ?: throw Exception(intl.urlChangedError(name))
+
+        val apiHeaders = headersBuilder()
+            .add("Accept", ACCEPT_JSON)
+            .build()
+
+        val jsonString = client.newCall(GET("$apiUrl/series/$mangaSlug", apiHeaders)).execute().body.string()
+
+        val result = jsonString.parseAs<HeanCmsSeriesDto>()
```

Comment:
Can use the `Response.parseAs` extension function directly.

```suggestion
        val detailsRequest = GET("$apiUrl/series/$mangaSlug", apiHeaders)
        val result = client.newCall(detailsRequest).execute()
            .parseAs<HeanCmsSeriesDto>()
```